sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Snippets/Sample07_MockEventProcessorHandlersTests.cs

```diff
@@ -0,0 +1,244 @@
+            bool checkpointUpdated = false;
+
+            async Task processEventHandler(ProcessEventArgs args)
+            {
+                try
+                {
+                    checkpointUpdated = false;
+
+                    await Application.ProcessEventAsync(args);
+
+                    string partition = args.Partition.PartitionId;
+
+                    int eventsSinceLastCheckpoint = partitionEventCount.AddOrUpdate(
+                        key: partition,
+                        addValue: 1,
+                        updateValueFactory: (_, currentCount) => currentCount + 1);
+
+                    if (eventsSinceLastCheckpoint >= EventsBeforeCheckpoint)
+                    {
+                        await args.UpdateCheckpointAsync();
+                        partitionEventCount[partition] = 0;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Application.HandleProcessingException(args, ex);
+                }
+            }
+
+            // Initialize the args and simulate checkpoint update method.
+
+            var partitionId = "0";
+            var eventBody = new BinaryData("This is a sample event body");
+            var eventData = EventHubsModelFactory.EventData(eventBody);
+            var partitionContext = EventHubsModelFactory.PartitionContext(partitionId);
+            var eventArgs = new ProcessEventArgs(partitionContext, eventData, _ => { checkpointUpdated = true; return Task.CompletedTask; });
+
+            for (int i = 1; i <= EventsBeforeCheckpoint; i++)
+            {
+                // Execute the handler and validate that the event handling has been checkpointed.
+
+                await processEventHandler(eventArgs);
+
+                if (i % EventsBeforeCheckpoint == 0)
+                    Assert.IsTrue(checkpointUpdated);
+                else
+                    Assert.IsFalse(checkpointUpdated);
+
+                Assert.That(partitionEventCount[partitionId], Is.EqualTo(i % EventsBeforeCheckpoint));
+            }
+
+            #endregion
+        }
+
+        /// <summary>
+        ///   Performs basic smoke test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ProcessErrorHandlerCancellationTrigger()
+        {
+            #region Snippet:EventHubs_Processor_Sample07_ProcessErrorHandlerCancellationTrigger
+
+            using var cancellationSource = new CancellationTokenSource();
+
+            // Applicaition handler cancels events processing after detection of the
+            // exact error with exact operation.
+
+            Task processErrorHandler(ProcessErrorEventArgs args)
+            {
+                if (args.Exception.Message.Equals("Example exception") && args.Operation.Equals("Example operation"))
+                {
+                    cancellationSource.Cancel();
```

Comment:
This is supposed to flow the other way.    The processor signals that cancellation should happen.   I'm a bit uncomfortable showing this flow; I fear it may be confusing.